app/models/agents/http_status_agent.rb

```diff
@@ -60,11 +61,12 @@ def receive(incoming_events)
     private
 
     def check_this_url(url)
-      if result = ping(url)
-        create_event payload: { 'url' => url, 'status' => result.status.to_s, 'response_received' => true }
-        memory['last_status'] = result.status.to_s
+      measured_result = TimeTracker.track { ping(url) }
```

Comment:
You have a point regarding the source of time. I think I'll swap it for a MONOTONIC one so we have the time only going forward.

The use of an object is basically to allow simple reutilization and not pollute the check code with the timing itself.